datadryad/dryad-repo
# Dash transfer -- initial version

This is the first version of the transfer mechanism between classic Dryad and New Dryad. It transfers basic metadata about a data package (title, authors, abstract). It can be run over a single data package, or the entire collection of data packages.

To test:
1. Ensure your maven settings file has appropriate values for 
  - default.dash.server = protocol and hostname of the target server (e.g. http://ryandash.datadryad.org)
  - default.dash.username = oauth application ID
  - default.dash.password = oauth application secret
2. Find the handle of an item you want to transfer
3. Call the tool from the command line, using the handle as the item's ID:
`/opt/dryad/bin/dspace curate -v -t transfertodash -i 10255/dryad.135814 -r -`

If you want to transfer all data packages, use the handle of the Data Packages collection (usually 10255/3).

Caveats:
- Although this PR includes a definition for a new metadata field, dashTransferDate, this field is not yet used. It will be used in a future update.
- The CSV written to stdout is intended to give basic feedback for someone watching the script run. It has been copied from the old DataPackageStats report, and currently contains some fields that do not yet have values. These values will be added in a future update.
